### PR TITLE
Gateway now also checks for exile implants in buckled mobs.

### LINKED ISSUE #9402: Exile implants can be bypassed
**Problem Description**:
It is possible to move someone from an away mission, to the station, even if they have an exile implant.

**What did you expect to happen**: 
I expected mobs with exile implants to never make it through the gateway.

**What happened instead**:
It is possible to use the way buckle mechanics work to effectively teleport an exile-implanted mob out of the gateway.

**Why is this bad/What are the consequences:**
This is arguably an exploit that nullifies the use of exile implants.

**Steps to reproduce the problem**:
1) Inside the gateway, implant a carbon mob with an exile implant, then buckle them to an office chair (chair/office/light). Force the chair through the gateway.
2) On station side, move the chair by any means. The buckled, exile-implanted mob will be teleported to the chair, station side, in a way that bypasses their exile implant.

**Fix idea**:
We could make it so that when an object is pushed through the gateway, it won't transit the gate if any attached mob has an exile implant.

### PR DESCRIPTION
**Pull request by alex-gh**
Fixes #9402

:cl: uc_guy
fix: Fixed exploit that allowed exiled mobs to return to the station
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/gateway.dm [hunk 1]
```diff
@@ -246,6 +246,8 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 	if(istype(M, /mob/living/carbon))
 		if(exilecheck(M)) return
 	if(istype(M, /obj))
+		if(M.can_buckle && M.has_buckled_mobs())
+			if(exilecheck(M.buckled_mob)) return
```
Comment (Birdtalon) [inline code/modules/awaymissions/gateway.dm]: The return should be on a newline.

Comment (alex-gh) [reply code/modules/awaymissions/gateway.dm]: Just following the style of the code around it, but sure, can do.

Comment (Birdtalon) [reply code/modules/awaymissions/gateway.dm]: Probably just old code, our contribution guide says that control statements should not have code on the same line as the statement so probably best to follow that to try and bring all code to the same standard :)

Comment (alex-gh) [reply code/modules/awaymissions/gateway.dm]: Should I fix all instances of it in this file? Or just the one I added?

Comment (Birdtalon) [reply code/modules/awaymissions/gateway.dm]: That's up to you really.

**Pull request merged**